doc/doxygen/headers/automatic_and_symbolic_differentiation.h

```diff
@@ -14,47 +14,389 @@
+ *   title     = {The Development of Hyper-Dual Numbers for Exact Second-Derivative Calculations},
+ *   booktitle = {49th {AIAA} Aerospace Sciences Meeting including the New Horizons Forum and Aerospace Exposition},
+ *   year      = {2011},
+ *   volume    = {886},
+ *   pages     = {124},
+ *   month     = {jan},
+ *   publisher = {American Institute of Aeronautics and Astronautics},
+ *   doi       = {10.2514/6.2011-886},
+ * }
+ * @endcode
+ *
+ * @code{.bib}
+ * @Manual{Walther2009a,
+ *   title     = {Getting Started with ADOL-C},
+ *   author    = {Walther, Andrea and Griewank, Andreas},
+ *   year      = {2009},
+ *   booktitle = {Combinatorial scientific computing},
+ *   doi       = {10.1.1.210.4834},
+ *   pages     = {181--202}
+ * }
+ * @endcode
+ *
+ * ### Explotation of the chain-rule
+ *
+ * In the most practical sense, any of the above categories exploit the chain-rule to compute the total
+ * derivative of a composite function. To perform this action, they typically use one of two mechanisms to
+ * compute derivatives, specifically
+ * - <em>forward-mode</em> (or <em>forward accumulation</em>) auto-differentation, or
+ * - <em>reverse-mode</em> (or <em>reverse accumulation</em>) auto-differentation.
+ * As a point of interest, the <em>optimal Jacobian accumulation</em>, which performs a minimal set of
+ * computations, lies somewhere between these two limiting cases. Its computation for a general composite
+ * function remains an open problem in graph theory.
+ *
+ * With the aid of the diagram below (it and some of the listed details courtesy of this
+ * <a href="https://en.wikipedia.org/wiki/Automatic_differentiation">Wikipedia article</a>),
+ *
+ * <div class="twocolumn" style="width: 80%">
+ *   <div class="parent">
+ *     <div class="img" align="center">
+ *       <img src="https://upload.wikimedia.org/wikipedia/commons/a/a4/ForwardAccumulationAutomaticDifferentiation.png"
+ *            alt="Forward mode automatic differentiation"
+ *            width="400">
+ *     </div>
+ *     <div class="text" align="center">
+ *       Forward mode automatic differentiation
+ *     </div>
+ *   </div>
+ *   <div class="parent">
+ *     <div class="img" align="center">
+ *       <img src="https://upload.wikimedia.org/wikipedia/commons/a/a0/ReverseaccumulationAD.png"
+ *            alt="Reverse mode automatic differentiation"
+ *            width="400">
+ *     </div>
+ *     <div class="text" align="center">
+ *       Reverse mode automatic differentiation
+ *     </div>
+ *   </div>
+ * </div>
+ *
+ * representing the calculation of the function
+ * \f[
+ *   f (\mathbf{x})
+ *     = x_{1} \times x_{2} + \sin (x_{1})
+ *   \quad ,
+ * \f]
+ * we will briefly describe what forward- and reverse- auto-differentiation are.
+ * Note that in the diagram, along the edges of the graph in text are the directional
+ * derivative of function \f$ w \f$ with respect to the i-th variable, represented by
```

Comment:
here and below it should be enough to do `$`, AFAIR dealii does some preprocessing. But check compiled page to be sure

Reply:
Yes, you are right. I will change this!